### PR TITLE
Database setup and migration

### PR DESCRIPTION
**Pull request by Shakir-Hussain12**
**For this PR, I :**
- Initialized rails project(blog_app) and setup Linters.
- Setup `postgres` as the working database.
- Created model migration to create tables using ERD.
- Added reference migrations for tables with foreign keys to match ERD.
- Added my migration to the actual database with `rails db:migrate`.

### CODE REVIEW

**Review by Ismailco — COMMENTED** [review]
Hi @Shakir-Hussain12, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has some linter errors. kindly make sure to fix the RSpec tests so the linter checks can pass

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by addod19** [comment]
### Hi @Shakir-Hussain12, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is invalid because the content of the README file is not for this current project, we are required to customized the project according to the project we work on

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: db/migrate/20230505010811_create_posts.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.string :author_id
+      t.string :title
+      t.string :text
```
Comment (codepantha) [inline db/migrate/20230505010811_create_posts.rb:6]: - [ ] For the post content, a `string` column would be insufficent. `:string` is mapped to `VARCHAR` and has a limit of 255 characters. 
As a general rule of thumb, use `:string` for short text input (username, email, password, titles, etc.) and use `:text` for longer expected input such as descriptions, post body, comment content etc.

File: db/migrate/20230505011116_create_comments.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.string :author_id
+      t.string :text
```
Comment (codepantha) [inline db/migrate/20230505011116_create_comments.rb:5]: - [ ] For the comment text, a `string` column would be insufficent. `:string` is mapped to `VARCHAR` and has a limit of 255 characters. 
As a general rule of thumb, use `:string` for short text input (username, email, password, titles, etc.) and use `:text` for longer expected input such as descriptions, post body, comment content etc.

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @Shakir-Hussain12 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Created data models 📌 
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @Shakir-Hussain12, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- [ ] No linters errors ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Created data models  ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**